.github/workflows/prof_correctness.yml

```diff
@@ -0,0 +1,70 @@
+name: Profiling correctness
+
+on:
+  - pull_request
+
+jobs:
+  prof-correctness:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.2'
+
+      - name: Restore cached Primes
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build profiler
+        run: |
+          sudo apt install -y clang
+          cd profiling
+          version_number=$(awk -F' = ' '$1 == "channel" { gsub(/"/, "", $2); print $2 }' rust-toolchain.toml)
+          curl https://sh.rustup.rs -sSf | sh -s -- --profile minimal -y --default-toolchain "$version_number"
+          cargo build --release
+
+      - name: Cache build dependencies
+        uses: actions/cache/save@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Run tests
+        run: |
+          php -d extension=target/release/libdatadog_php_profiling.so -v
+          mkdir profiling/tests/correctness/allocations/
```

Comment:
As we still have the nightly that tests the delivered binaries, this is fine.
As discussed we could think of ways to factorize some of the test setups.